mdsuite/graph_modules/molecular_graph.py

```diff
@@ -247,14 +256,224 @@ def check_a_in_b(a, b):
         for item in del_list:
             molecules.pop(item)
 
-        if n_molecules is None:
-            return molecules
+        return molecules
+
+    def _perform_isomorphism_tests(self):
+        """
+        Run isomorphism checks to determine whether or not the graphs computed are
+        correct.
+
+        Currently runs the following tests:
+
+        1. Checks that the number of decomposed graphs is equal to the number of
+           expected molecules.
+        2. Checks that the number of particles of each constituent species for each
+           molecule matches that given by the SMILES string or the user provided
+           reference data.
+
+        Returns
+        -------
+
+        """
+        # amount of molecules test
+        self._amount_isomorphism_test()
+        # groups equality test
+        self._molecule_group_equality_isomorphism_test()
+
+    def _amount_isomorphism_test(self):
+        """
+        Test that the amount of computed molecules is equal to the expected amount.
+
+        Returns
+        -------
+        Returns nothing, raises a value error if condition is not met.
+        """
+        log.info("Performing molecule number isomorphism test.")
```

Comment:
That is a lot of logging. I think `passed` would be enough for the info and the rest could log to debug?

Reply:
I thought about it, I also think that the molecule mapping is quite a strenuous set of operations. Each test is checking something different and they can start to take time for very large systems, e.g, with thousands of molecules. If we add the final full graph isomorphism test, which we could theoretically, that would be an N^2 operation and require a ton of time. Therefore, I left them is a info logs so users know what stage of the mapping they are up to. For small systems this will just be a lot of output very fast but for large systems there will be laps between them